plugins/out_kinesis_firehose/firehose_api.c

```diff
@@ -235,11 +236,10 @@ static int process_event(struct flb_firehose *ctx, struct flush *buf,
         time_key_ptr += 3;
         tmp_size = buf->tmp_buf_size - buf->tmp_buf_offset;
         tmp_size -= (time_key_ptr - tmp_buf_ptr);
-        len = strftime(time_key_ptr, tmp_size, ctx->time_key_format, &time_stamp);
-        if (len <= 0) {
-            /* ran out of space - should not happen because of check above */
-            return 1;
-        }
+        /* merge out_buff to time_key_ptr */
+        len = snprintf(time_key_ptr, strlen(out_buff) + 1,
```

Comment:
I think using snprintf is not necessary and might not be most efficient- we are just copying a string to a buf- memcpy or strncpy is for that